data/src/main/java/com/teamagam/gimelgimel/data/message/poller/MessageLongPoller.java

```diff
@@ -44,18 +40,20 @@ public MessageLongPoller(GGMessagingAPI messagingAPI, MessageDataMapper messageD
   public Observable poll() {
     long synchronizedDateMs = mPrefs.getLong(Constants.LATEST_MESSAGE_DATE_KEY);
 
-    return poll(synchronizedDateMs).map(this::getMaximumDate)
+    return poll(synchronizedDateMs)
+        .toFlowable(BackpressureStrategy.BUFFER)
+        .map(this::getMaximumDate)
         .filter(newSynchronizationDate -> newSynchronizationDate != NO_NEW_MESSAGES)
         .doOnNext(newSynchronizationDate -> mPrefs.setPreference(Constants.LATEST_MESSAGE_DATE_KEY,
             newSynchronizationDate))
         .onErrorResumeNext(throwable -> {
           if (isPollingException(throwable)) {
-            return Observable.just(synchronizedDateMs);
+            return Flowable.just(synchronizedDateMs);
           } else {
-            return Observable.error(throwable);
+            return Flowable.error(throwable);
           }
         })
-        .onBackpressureBuffer();
+        .toObservable();
```

Comment:
You do nothing by casting back to `Observable`.
It should remain `Flowable` in the entire processing stage.